addon/components/column-chart/component.js

```diff
@@ -428,34 +432,76 @@ export default BaseChartComponent.extend({
 
-        let values = [];
-        let groups = this.getWithDefault('group', []);
+        let values;
+        const groups = this.getWithDefault('group', []);
         groups.forEach((g, index) => {
             if (index === this.get('seriesMaxMin')) {
                 values = g.all().map(gElem => gElem.value);
             }
         });
 
-        for (let i = 0; i < bars.length; i++) {
-            if (!values[i]) {
-                continue;
-            }
+        /* auto adjust when to hide data value of a bar
+         * if labels overlap.
+         */
+        if (this.get('labelOptions.labelCollisionResolution') === 'auto' && bars.length > 1) {
+            const barWidth = Number(d3.select(bars[0]).attr('width'));
+            const barGap = Math.abs(Number(d3.select(bars[0]).attr('x')) - Number(d3.select(bars[1]).attr('x'))) - barWidth;
+
+            // how many labels to skip.
+            let skipLabels = 0;
+            for (let i = 0; i < bars.length; i++) {
+                if (!values[i] || skipLabels > 0) {
+                    skipLabels--; // !values[i] does not render the bar, the gap increases, it is equivalent to skipping a label.
+                    continue;
+                }
 
-            gLabels.append('text')
-                .text(() => formatter(values[i]))
-                .attr('x', () => +d3.select(bars[i]).attr('x') + (d3.select(bars[i]).attr('width') / 2))
-                .attr('y', Math.max(12, isBottomLabelPosition ? maxLabelYHeight + 12 : maxLabelY - 2))
-                .attr('text-anchor', 'middle')
-                .attr('font-size', '12px')
-                .attr('fill', this.getWithDefault('colors', [])[this.get('seriesMaxMin')])
-                .attr('class', 'data-text');
+                const label = gLabels.append('text')
+                    .text(() => formatter(values[i]))
+                    .attr('x', () => d3.select(bars[i]).attr('x'))
+                    .attr('y', Math.max(12, isBottomLabelPosition ? maxLabelYHeight + 12 : maxLabelY - 2))
+                    .attr('font-size', '12px')
+                    .attr('fill', this.getWithDefault('colors', [])[this.get('seriesMaxMin')])
+                    .attr('class', 'data-text')
+                    .attr('id', `data-text-${i}`);
+
+                const labelDimensions = label.node().getBBox();
+                const maxLabelDimensions = maxLabel.node().getBBox();
+                const minLabelDimensions = minLabel.node().getBBox();
+                const leftMostAmongMaxAndCurrentLabel = labelDimensions.x - maxLabelDimensions.x < 0 ? labelDimensions : maxLabelDimensions;
+                const leftMostAmongMinAndCurrentLabel = labelDimensions.x - minLabelDimensions.x < 0 ? labelDimensions : minLabelDimensions;
+                
+                // check if label collides with min/max label
+                if (Math.abs(labelDimensions.x - maxLabelDimensions.x) < leftMostAmongMaxAndCurrentLabel.width
```

Comment:
I think this logic checks out. Could you add a few tests for this as well? A few cases to consider:

```
regular label overlapping max label from left
regular label not overlapping max label from left

max label overlapping a regular label from the left
max label not overlapping regular label from the left
```